src/Microsoft.OData.Core/UriParser/ODataUriParser.cs

```diff
@@ -607,6 +603,26 @@ private void InitQueryOptionDic()
             }
         }
 
+        /// <summary>

```

Comment:
Do you think we should add a public API in the ODataUriParser to get the custom query option nodes?
Same as 'ParseFilter()'?

Reply:
I thought of that, but we already have a CustomQueryOptions public property which returns the key/value pairs (which I would expect is the same as what ParseCustomQueryOption() would do.)